src/Sushi.Mediakiwi/Encryption/EncryptionHelper.cs

```diff
@@ -0,0 +1,145 @@
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Data.Notification.InsertOne(nameof(EncryptionHelper), ex.Message);
+            }
+
+            return result;
+        }
+
+        public string DecryptString(string cipherText)
+        {
+            // Declare the string used to hold  
+            // the decrypted text.  
+            string result = null;
+
+            try
+            {
+                // generate the key from the shared secret and the salt  
+                using (Rfc2898DeriveBytes key = new Rfc2898DeriveBytes(_encryptionKey, _salt))
+                {
+                    // Create the streams used for decryption.  
+                    byte[] bytes = Convert.FromBase64String(cipherText);
+                    using (MemoryStream msDecrypt = new MemoryStream(bytes))
+                    {
+                        // Create a RijndaelManaged object  
+                        // with the specified key and IV.  
+                        using (RijndaelManaged aesAlg = new RijndaelManaged())
+                        {
+                            aesAlg.Key = key.GetBytes(aesAlg.KeySize / 8);
+                            // Get the initialization vector from the encrypted stream  
+                            aesAlg.IV = ReadByteArray(msDecrypt);
+                            // Create a decrytor to perform the stream transform.  
+                            ICryptoTransform decryptor = aesAlg.CreateDecryptor(aesAlg.Key, aesAlg.IV);
+                            using (CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
+                            {
+                                using (StreamReader srDecrypt = new StreamReader(csDecrypt))
+                                {
+                                    // Read the decrypted bytes from the decrypting stream  
+                                    // and place them in a string.  
+                                    result = srDecrypt.ReadToEnd();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Data.Notification.InsertOne(nameof(EncryptionHelper), ex.Message);
+            }
+
+            return result;
+        }
+
+        private static byte[] ReadByteArray(Stream s)
+        {
+            byte[] rawLength = new byte[sizeof(int)];
+            if (s.Read(rawLength, 0, rawLength.Length) != rawLength.Length)
+            {
+                throw new SystemException("Stream did not contain properly formatted byte array");
+            }
+
+            byte[] buffer = new byte[BitConverter.ToInt32(rawLength, 0)];
+            if (s.Read(buffer, 0, buffer.Length) != buffer.Length)
+            {
+                throw new SystemException("Did not read byte array properly");
+            }
```

Comment:
Why throw SystemExceptions? Is it maybe better to use InvalidOperationException?
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types

Reply:
Agreed, will throw an argument exception for the first and an InvalidOperationException for the second